rfd/0051-x11-forwarding.md

```diff
@@ -0,0 +1,205 @@
+in the session. This is where understanding `trusted` vs `untrusted` forwarding is 
+very important, which have their own sections below.
+
+#### Client Authentication - X11 Channel Request
+
+When the SSH client receives an `x11` channel request, it must authenticate the server
+before providing access to the client's local X Server. To do this, it will use the
+X authorization data it sent in the original `x11-req`
+
+Note: The X authorization data sent in the `x11-req` is fake X authorization data
+created by the server. 
+
+First, the client sends fake X authorization data to the server in the `x11-req` request.
+As detailed in the SSH RFC, this data includes an X authorization protocol and a random
+cookie of that protocol. We use the `MIT-MAGIC-COOKIE-1` protocol, which calls for a random
+128 bit hexadecimal encoded cookie. This is the simplest and most common protocol.
+
+When the client receives an `x11` SSH channel request from the server, it mimics and
+X Server by scanning the connection for its X authorization packet, the first portion
+of the request. If the X authorization packet includes the protocol and cookie sent by
+the client, then the server will be trusted.
+
+Note: The X authorization data used in this step is fake and does not actually provide
+access to the client's XServer. It's essential that an XServer's X authorization data
+is never shared outside of the device, or else the XServer can be compromised.
+
+#### Client X Server Authorization
+
+To complete X11 forwarding, the client must start forwarding between the `x11` channel
+and the local X Server. However, the X authorization data in the `x11` channel request
+was the fake data we created. So before forwarding the X request to the X Server, the
+SSH client replaces the fake data with real X authorization data. This data will either
+be trusted xauth data, or untrusted xauth data, depending on what the client requested.
+
+##### Untrusted X11 Forwarding
+
+Untrusted X authorization data must be generated through `xauth` in order to be recognized
+by the X Server. To do this, the client runs `xauth generate <$DISPLAY> untrusted <timeout?>`.
+Any X Client request using this untrusted token will be be subject to the X Security extension,
+granting it fewer privileges than usual. 
+
+##### Trusted X11 Forwarding
+
+In trusted X11 forwarding, we can rely on the client's local environment. If we fill in 
+the X authorization packet in the request with a random cookie, the X Server will just
+ignore it and connect as if it received the request locally.
+
+### tsh UX
+
+`tsh ssh -X` or `tsh ssh -Y` can be used to request a session with `trusted` X11 forwarding.
+However, if `tsh ssh -X` is used with the `-oForwardX11Trusted=yes` flag, it will instead
+request `untrusted` forwarding, while `tsh ssh -Y` would still request `trusted`.
+
+This may seem unintuitive, and it is, but this is the same UX as OpenSSH client. In OpenSSH,
```

Comment:
This doesn't seem like a good enough reason to me.

I wonder if it would be worth creating a more intuitive UX, and also supporting the openssh flags for compatibility.

Reply:
Agreed, updated the UX and added an `deviations from openssh` section to explain the change.